### ISSUE TITLE
Tuckey "forward" toType not working to CMS content destination

### ISSUE THREAD

**Issue by chrismccracken** [dotCMS/core#9665]
Version 3.5.1

UrlRewriteFilter filter-mapping moved to top of chain.

```
public class Activator extends GenericBundleActivator {

    @SuppressWarnings ("unchecked")
    public void start ( BundleContext context ) throws Exception {

        initializeServices( context );

        NormalRule debugRule = new NormalRule();
        debugRule.setFrom( "^/en-CA/(.*?)" );
        debugRule.setToType( "forward" );
        debugRule.setTo( "/$1" );

        addRewriteRule( debugRule );

    }

    public void stop ( BundleContext context ) throws Exception {

        unregisterServices( context );
    }

}
```

"redirect" type works properly, "forward" throws a 404

**Comment by chrismccracken** [comment]
Re: https://my.dotcms.com/tickets/detail.dot?id=67d6abfc-5d62-49fb-a204-86efc55ee689

**Comment by cfalzone** [comment]
Do you guys have a hotfix / workaround or anything for this issue in version 3.6.2? 

We have this same problem on our upgrade, redirects work, but forwards do not.  I also tried moving  the URLRewriteFilter to the top of the chain in web.xml but it didn't solve the issue.

**Comment by jtesser** [comment]
hmm must have something to do with the updated CMSFilter 

I assume these are pages as content pages and not files?  But maybe happens for both?  url map content?

**Comment by cfalzone** [comment]
The detail page here is a content page (all our pages are now).  

The rule in question is:

```
        NormalRule rule = new NormalRule();
        rule.setName("JobsURLMap");
        rule.setFrom("^/find-work/([0-9]+)/?$");
        rule.setTo("/find-work/details.htm?jobId=$1");
        rule.setToType("forward");
        rule.addCondition(aqHostCond);
        addRewriteRule(rule);
```

Had the same problem in 3.2.3 but moving the URLRewriteFilter above the CMSFilter fixed in 3.2.3 but does not appear to help in 3.6.2.

**Comment by wezell** [comment]
This will be addressed with the changes here #12315

**Comment by cfalzone** [comment]
For those looking for a work-around:  https://gist.github.com/chrismccracken/77d11cec98fb636452213c9fb69c568e
